pkg/antctl/raw/supportbundle/command.go

```diff
@@ -692,9 +702,167 @@ func processResults(resultMap map[string]error, dir string) error {
+	}(); err != nil {
+		return err
+	}
+	if podRef == nil {
+		return fmt.Errorf("no podRef found in antrea controllerInfo")
+	}
+	pod, err := k8sClient.CoreV1().Pods(podRef.Namespace).Get(ctx, podRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if err := downloadPodLogs(ctx, k8sClient, pod.Namespace, pod.Name, k8s.GetPodContainerNames(pod), tmpDir); err != nil {
+		return err
+	}
+	return packPodLogs(pod, dir, tmpDir)
+}
+
+func downloadFallbackAgentBundleFromKubernetes(ctx context.Context, antreaClientset antrea.Interface, k8sClient kubernetes.Interface, failedNodes []string, dir string) error {
+	agentInfoList, err := antreaClientset.CrdV1beta1().AntreaAgentInfos().List(ctx, metav1.ListOptions{ResourceVersion: "0"})
+	if err != nil {
+		return err
+	}
+
+	agentInfoMap := map[string]v1beta1.AntreaAgentInfo{}
+	for _, agentInfo := range agentInfoList.Items {
+		agentInfoMap[agentInfo.Name] = agentInfo
+	}
+	pods, err := k8sClient.CoreV1().Pods("kube-system").List(ctx, metav1.ListOptions{
+		ResourceVersion: "0",
+		LabelSelector:   "app=antrea,component=antrea-agent",
+	})
+	if err != nil {
+		return err
+	}
+	failedNodeSet := sets.NewString(failedNodes...)
+	var errors []error
+	for _, pod := range pods.Items {
+		if !failedNodeSet.Has(pod.Spec.NodeName) {
+			continue
+		}
+		if err := func() error {
+			tmpDir, err := afero.TempDir(defaultFS, "", "bundle_tmp_")
+			if err != nil {
+				return err
+			}
+			defer defaultFS.RemoveAll(tmpDir)
+			if agentInfo, ok := agentInfoMap[pod.Spec.NodeName]; ok {
+				data, err := yaml.Marshal(agentInfo)
+				if err != nil {
+					return err
+				}
+				if err = afero.WriteFile(defaultFS, filepath.Join(tmpDir, "agentinfo"), data, 0644); err != nil {
+					return err
+				}
+			}
+			err = downloadPodLogs(ctx, k8sClient, pod.Namespace, pod.Name, k8s.GetPodContainerNames(&pod), tmpDir)
+			if err != nil {
+				return err
+			}
+			return packPodLogs(&pod, dir, tmpDir)
+		}(); err != nil {
+			errors = append(errors, err)
+		}
+	}
+	return utilerror.NewAggregate(errors)
+}
+
+func packPodLogs(pod *corev1.Pod, dir string, logsDir string) error {
+	prefix := "agent_"
+	if strings.Contains(pod.Name, "controller") {
+		prefix = "controller_"
+	}
+	gzFileName := filepath.Join(dir, prefix+pod.Spec.NodeName+".tar.gz")
+	f, err := defaultFS.Create(gzFileName)
+	if err != nil {
+		return err
+	} else {
+		defer f.Close()
+		_, err := compress.PackDir(defaultFS, logsDir, f)
+		return err
+	}
+}
+
+func downloadPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace string, podName string, containers []string, tmpDir string) error {
+	downloadContainerLogs := func(containerName string) error {
+		containerDirName, _ := strings.CutPrefix(containerName, "antrea-")
+		containerLogDir := filepath.Join(tmpDir, "logs", containerDirName)
+		err := os.MkdirAll(containerLogDir, 0755)
+		if err != nil {
+			return err
+		}
+		fileName := filepath.Join(containerLogDir, containerName+".log")
```

Comment:
I may be missing something, but this doesn't look right to me. For an agent, it seems we will end up with `logs/agent/antrea-agent.log` and `logs/ovs/antrea-ovs.log`. Is that intentional?

Reply:
https://github.com/antrea-io/antrea/pull/3659#issuecomment-1137061573 

well,  the directory structure should be similar to  the normal case.